src/Scripts.pas

```diff
@@ -0,0 +1,1638 @@
+    result.handled := result.handled or res;
+  end;
+end;
+
+function TScriptsHandler.getVecVar(name: string): TD3DXVector3;
+var
+  i, n:integer;
+begin
+  n:=length(_vecVars);
+  result:=D3DXVector3Zero;
+  for i:=0 to n - 1 do
+    if _vecVars[i].name = name then
+    begin
+      result:=_vecVars[i].pos;
+      exit;
+    end;
+
+  //ez mi�rt nem script error lel debug loife
+  multisc.chats[addchatindex].uzenet := 'No such vector variable: ' + name;
+  addchatindex:=addchatindex + 1;
+end;
+
+function TScriptsHandler.getNumVar(name: string): single;
+var
+  i, n:integer;
+begin
+  n:=length(_numVars);
+  result:=0;
+  for i:=0 to n - 1 do
+    if _numVars[i].name = name then
+    begin
+      result:=_numVars[i].num;
+      exit;
+    end;
+
+  //ez mi�rt nem script error lel debug loife part 2
+  multisc.chats[addchatindex].uzenet := 'No such numeric variable: ' + name;
+  addchatindex:=addchatindex + 1;
+end;
+
+procedure TScriptsHandler.varCompare(args: TStringArray);
+var
+  argnum, i, j:integer;
+  v1, v2:single;
+  truth:boolean;
+  op:string;
+  a, b:TStringArray;
+begin
+  argnum:=length(args);
+
+  //find the comparator
+  for i:=0 to high(args) do
+  begin
+
+    if ((args[i] <> '=') or (args[i] <> '>') or (args[i] <> '<') or
+        (args[i] <> '!=') or (args[i] <> '>=') or (args[i] <> '<=')) then
+    begin
+      continue;
+    end;
+
+    op := args[i];
+    setlength(a, i);
+    setlength(b, argnum - i - 1);
+
+    for j:=0 to i - 1 do
+      a[j] := args[j];
+
+    for j:=0 to argnum - i - 2 do
+      b[j] := args[i + 1 + j];
+
+    v1:=compute(a)[0];
+    v2:=compute(b)[0];
+
+    truth := false;
+
+    if (op = '=') and (v1 = v2) then truth := true;
+    if (op = '<') and (v1 < v2) then truth := true;
+    if (op = '>') and (v1 > v2) then truth := true;
+    if (op = '!=') and (v1 <> v2) then truth := true;
+    if (op = '<=') and (v1 <= v2) then truth := true;
+    if (op = '>=') and (v1 >= v2) then truth := true;
+
+    inc(_scriptdepth);
+    _scriptevaling[_scriptdepth] := truth;
+
+    exit;
+  end;//for
+
+end;
+
+function TScriptsHandler.computeVecs(words: TStringArray): TD3DXVector3;
+var
+  tmp:TSingleArray;
+  tmp2:TStringArray;
+  i, n:integer;
+begin
+  n:=length(words);
+  setLength(tmp2, n);
+
+  for i:=0 to n - 1 do
+    tmp2[i]:=words[i];
+
+  tmp:=compute(tmp2);
+  result:=D3DXVector3(tmp[0], tmp[1], tmp[2]);
+end;
+
+function TScriptsHandler.computeNums(words: TStringArray): single;
+var
+  tmp:TSingleArray;
+  tmp2:TStringArray;
+  i, n:integer;
+begin
+  n:=length(words);
+  setLength(tmp2, n);
+
+  for i:=0 to n - 1 do
+    tmp2[i]:=words[i];
+
+  tmp:=compute(tmp2);
+  result:=tmp[0];
+end;
+
+function TScriptsHandler.compute(words: TStringArray): TSingleArray;
+var
+  i, j, n:integer;
+  tmp:TD3DXVector3;
+  error:Integer;
+  numnum, varnum:integer;
+  nums, vars:array[0..2] of single;
+  operator:char;
+  value:single;
+  r:TSingleArray;
+begin
+  // ! a vektor v�ltoz�
```

Comment:
TODO rewrite old comments in English so less stuff breaks when encoding hits the fan